### PR TITLE
Support string type for infinity norm in attacks

### LINKED ISSUE #465: Allow norm to be passed to attacks as string argument
**Is your feature request related to a problem? Please describe.**
Armory current reads kwargs for attacks from a json file. Most input kwargs are easily serialized 
to json, however the norm parameter (used for the FastGradientMethod attack and
ProjectedGradientDescent attack) may take np.inf as an argument. It would be convenient 
if the attacks also accepted a string, e.g. "inf" as an argument and contained logic in the 
attack class to map this string argument to np.inf.

**Describe the solution you'd like**
Update the logic in the FGSM and other attacks in the [constructor](https://github.com/IBM/adversarial-robustness-toolbox/blob/3cfa6de196cb32a3efafab2ff6bbf44247c9ddbd/art/attacks/evasion/fast_gradient.py#L97) and/or [set_params](https://github.com/IBM/adversarial-robustness-toolbox/blob/3cfa6de196cb32a3efafab2ff6bbf44247c9ddbd/art/attacks/evasion/fast_gradient.py#L244)

### PR DESCRIPTION
**Pull request by beat-buesser**
Signed-off-by: Beat Buesser <beat.buesser@ie.ibm.com>

# Description

This pull request add support for string type to define infinity norms in attacks.

Fixes #465 

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/attacks/evasion/fast_gradient.py [hunk 1]
```diff
@@ -250,8 +250,8 @@ def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> n
 
     def _check_params(self) -> None:
         # Check if order of the norm is acceptable given current implementation
-        if self.norm not in [np.inf, int(1), int(2)]:
-            raise ValueError("Norm order must be either `np.inf`, 1, or 2.")
+        if self.norm not in ["inf", np.inf, int(1), int(2)]:
```
Comment (minhitbk) [inline art/attacks/evasion/fast_gradient.py:253]: Could you make this line to be consistent with in other files:  
if self.norm not in [1, 2, np.inf, "inf"]:

File: art/attacks/evasion/hop_skip_jump.py [hunk 2]
```diff
@@ -542,8 +544,8 @@ def _interpolate(current_sample: np.ndarray, original_sample: np.ndarray, alpha:
 
     def _check_params(self) -> None:
         # Check if order of the norm is acceptable given current implementation
-        if self.norm not in [np.inf, int(2)]:
-            raise ValueError("Norm order must be either `np.inf` or 2.")
+        if self.norm not in ["inf", np.inf, int(2)]:
```
Comment (minhitbk) [inline art/attacks/evasion/hop_skip_jump.py:547]: To be consistent with other files:
if self.norm not in [2, np.inf, "inf"]:

File: art/attacks/evasion/projected_gradient_descent/projected_gradient_descent.py [hunk 3]
```diff
@@ -178,8 +178,8 @@ def set_params(self, **kwargs) -> None:
 
     def _check_params(self) -> None:
         # Check if order of the norm is acceptable given current implementation
-        if self.norm not in [np.inf, int(1), int(2)]:
-            raise ValueError("Norm order must be either `np.inf`, 1, or 2.")
+        if self.norm not in ["inf", np.inf, int(1), int(2)]:
```
Comment (minhitbk) [inline art/attacks/evasion/projected_gradient_descent/projected_gradient_descent.py:181]: For consistency:
if self.norm not in [1, 2, np.inf, "inf"]:

File: art/utils.py [hunk 4]
```diff
@@ -162,13 +162,13 @@ def wrapper(*args, **kwargs):
 # ----------------------------------------------------------------------------------------------------- MATH OPERATIONS
 
 
-def projection(values: np.ndarray, eps: float, norm_p: Union[int, float]) -> np.ndarray:
+def projection(values: np.ndarray, eps: float, norm_p: Union[int, str]) -> np.ndarray:
```
Comment (minhitbk) [inline art/utils.py:165]: Why don't you count `float` here as in other places?

**Review by minhitbk — APPROVED** [review]
Hi @beat-buesser , the changes in general look good to me, there are only a few minor comments that you can choose to either revise or not. They are not very important.

File: art/attacks/evasion/fast_gradient.py (hunk 1 above)
Comment (beat-buesser) [reply art/attacks/evasion/fast_gradient.py]: Fixed.

File: art/attacks/evasion/projected_gradient_descent/projected_gradient_descent.py (hunk 3 above)
Comment (beat-buesser) [reply art/attacks/evasion/projected_gradient_descent/projected_gradient_descent.py]: Fixed.

File: art/utils.py (hunk 4 above)
Comment (beat-buesser) [reply art/utils.py]: I think it just got forgotten, I have added it back for `np.inf`.

**Comment by beat-buesser** [comment]
Hi @minhitbk Thank you for your review! I have fixed everything in my last two commits.

**Pull request merged**